File: lib/src/models/pagination_result.dart
import 'package:honey_and_thyme/src/models/parsable.dart';

class PaginationResult<T extends Parsable> {
  List<T>? results;
  int? pageIndex;
  int? pageSize;
  int? pageCount;
  int? totalCount;

  PaginationResult({
    this.results,
    this.pageIndex,
    this.pageSize,
    this.pageCount,
    this.totalCount,
  });
}

File: honey-and-thyme/src/lib/types/paginationResult.ts
export default interface PaginationResult<T> {
  results?: T[] | null;
  /** Format: int32 */
  pageIndex?: number;
  /** Format: int32 */
  pageSize?: number;
  /** Format: int32 */
  pageCount?: number;
  /** Format: int32 */
  totalCount?: number;
}
